Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -1370,6 +1387,77 @@ void qSlicerSaveDataDialogPrivate::onItemChanged(QTableWidgetItem* widgetItem)
   }
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerSaveDataDialogPrivate::validateFilePathLength()
+{
+  ctkPathLineEdit* pathLineEdit = qobject_cast<ctkPathLineEdit*>(this->sender());
+  if (!pathLineEdit)
+  {
+    return;
+  }
+
+  int rowIndex = this->FileWidget->indexAt(pathLineEdit->pos()).row();
+  this->validateFilePathLength(rowIndex);
+}
+
+//-----------------------------------------------------------------------------
+void qSlicerSaveDataDialogPrivate::validateFilePathLength(int row)
+{
+  vtkMRMLScene* mrmlScene = this->MRMLScene;
+  if (!mrmlScene)
+  {
+    return;
+  }
+
+  QTableWidgetItem* fileNameItem = this->FileWidget->item(row, FileNameColumn);
+  if (!fileNameItem)
+  {
+    return;
+  }
+
+  QFileInfo fileInfo = this->file(row);
+  QString fileName = fileInfo.fileName();
+
+  // Determine current file extension
+  vtkObject* objectToSave = mrmlScene;
+  QString nodeID = fileNameItem->data(Self::UIDRole).toString();
+  if (!nodeID.isEmpty())
+  {
+    objectToSave = qSlicerSaveDataDialogPrivate::getNodeByID(nodeID.toUtf8().data(), mrmlScene);
+    if (!objectToSave)
+    {
+      qCritical() << Q_FUNC_INFO << " failed: node not found by ID " << qPrintable(nodeID);
+    }
+  }
+
+  qSlicerCoreIOManager* coreIOManager =
+    qSlicerCoreApplication::application()->coreIOManager();
+  QString extension = coreIOManager->extractKnownExtension(fileName, objectToSave);
+
+  // Check if the file name is too long
+  QStringList pathLengthWarningMessages;
+  QString shortFileName = qSlicerCoreIOManager::forceFileNameMaxLengthExtension(fileName, extension.length());
+  if (shortFileName != fileName)
+  {
+    pathLengthWarningMessages << qSlicerSaveDataDialog::tr("File name may be too long.");
+  }
+
+  int maxPathLength = 260;
```

Comment:
The limit should be lower, around 200, but I think we should only log this warning if saving failed.

Reply:
Done.